portfolio/src/main/webapp/script.js

```diff
@@ -12,17 +12,58 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+var /** number */ slideIndexManual = 1;
+var /** number */ slideIndexAutomatic = 1;
+manualSlideShowForGallery(slideIndexManual);
+automaticSlideShowForBlog();
+
 /**
- * Adds a random greeting to the page.
+ * Increments/decrements the slide currently being showcased on the
+ * gallery and switches to a new slide. 
+ * @param {number} whichButtonClick
  */
-function addRandomGreeting() {
-  const greetings =
-      ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!'];
+function switchSlides(whichButtonClick) {
+  manualSlideShowForGallery(slideIndexManual += whichButtonClick);
+}
 
-  // Pick a random greeting.
-  const greeting = greetings[Math.floor(Math.random() * greetings.length)];
+/**
+ * Based on the current index of the slideshow and which button was clicked,
+ * this function correctly shows the required slideshow and hides those not
+ * needed. 
+ * @param {number} whichSlideToShow
+ */
+function manualSlideShowForGallery(whichSlideToShow) {
+  var /** ?HTMLCollection */ gallerySlides =
+      document.getElementsByClassName("gallery-slides");
+  var /** ?number */ gallerySlidesIndex;
+  if (whichSlideToShow > gallerySlides.length) {
```

Comment:
A more elegant way of doing this is: slideIndexManual = whichSlide % gallerySlides.length;